### ISSUE TITLE
add off canvas push support ?

### ISSUE THREAD

**Issue by crapthings** [picturepan2/spectre#371]

**Comment by picturepan2** [comment]
@crapthings By adding a new class?

**Comment by Xeoncross** [comment]
If this is a request for adding a class (via JavaScript) to toggle the off-canvas element, then I'm all for that. Angular/Vue/React already do this well.

**Comment by laurentpayot** [comment]
@Xeoncross isnt the `active` class doing that already?

"push" support means pushing the page content to the right when opening the sidebar,  but maybe I'm wrong…

**Comment by Xeoncross** [comment]
@laurentpayot you are right. Here is a [simple example of pure react + spectre.css off-canvas using the "active" class](https://codesandbox.io/s/y2n91z45k1)
